### ISSUE TITLE
Contextual Gestures Don't Accumulate

### ISSUE THREAD

**Issue by unicorndreams** [marklieberman/foxygestures#54]
In Opera, contextual gesture could be accumulated. For example, DOWN gesture could either open a link in a new background tab or just open a new tab, if the gesture didn't start on a link, i.e, Opera didn't forbid the same gesture to be applied to different commands, as long as the commands didn't share the same scope. Of course, there may be cases in which two contexts overlap, for example, a video that is wrapped by a &lt;A HREF&gt; tag, if there is the same gesture assigned to open the link in a new window and to download the video, Foxy Gestures should prioritise one (or not, just pick the first action that matches, let the user be aware that accumulating actions on the same gesture may generate random effects depending on context).

**Comment by marklieberman** [comment]
I came from FireGestures, where stacking gestures was not a feature. As a result, I wrote a lot of code assuming a unique mapping of gesture to command. It would be a decent rewrite to a lot of the background and options UI code to change it now. I'm afraid I'm just not willing to commit to implementing this type of feature at this time.

I assume you left the latest review on AMO, so you already know you can kind of get there using a user script. I'm not sure if you saw #40, but you can call the built in commands from your user script.

**Comment by unicorndreams** [comment]
I understand, it was just an idea. On FireGestures, I had two custom scripts to mimic Opera's behaviour (dating back in the time I migrated from that browser), one for opening links on new tabs and another to open in new windows. I checked that post, and it really helped. I spent some time trying to figure out how to write the scripts, but I finally did it (although, not sure if this was the most efficient way to do it, but it works!) I'm posting it just in case someone else would want it:

`if (data.element.linkHref)
{
    executeInBackground((data) =>
    {
        commandOpenLinkInNewTab(data);
    }, [data])
}
else
{
    executeInBackground((data) =>
    {
        commandNewTab(data);
    }, [data]);
}`
